pointers.go

```diff
@@ -15,31 +15,58 @@ func mapReusedPointers(v reflect.Value) ptrmap {
 }
 
 // A map of pointers.
-type (
-	ptrinfo struct {
-		order int
+type ptrinfo struct {
+	order int
+}
+
+type ptrkey struct {
+	p uintptr
+	t string
+}
+
+func ptrkeyFor(v reflect.Value) (k ptrkey) {
+	k.p = v.Pointer()
+	for v.Kind() == reflect.Ptr {
+		v = v.Elem()
 	}
-	ptrmap map[uintptr]ptrinfo
-)
+	if v.IsValid() {
+		k.t = v.Type().Name()
+	}
```

Comment:
What about using `v.Type()` as a key instead of the name? The documentation says:

> Type values are comparable, such as with the == operator, so they can be used as map keys. Two Type values are equal if they represent identical types.
>
> https://golang.org/pkg/reflect/#Type

Reply:
Good point, will change.